packages/iov-ledger-bns/src/app.ts

```diff
@@ -13,16 +13,19 @@ const cmdPubkeyWithPath = 5;
 const cmdAppVersion = 0xca;
 
 export function getPublicKey(transport: TransportNodeHid): Promise<Uint8Array> {
-  return sendChunks(transport, appCode, cmdPubkey, new Uint8Array([]));
+  return sendChunks(transport, appCode, cmdPubkey, new Uint8Array([0]));
 }
 
 export function getPublicKeyWithIndex(transport: TransportNodeHid, i: number): Promise<Uint8Array> {
   const pathComponent = Slip10RawIndex.hardened(i).asNumber();
-  return sendChunks(transport, appCode, cmdPubkeyWithPath, encodeUint32(pathComponent));
+  let pathPart = encodeUint32(pathComponent);
+  pathPart = [pathComponent.length, ...pathPart];
```

Comment:
Does this compile? What you want is

`const pathPart = new Uint8Array([pathComponent.length, ...encodeUint32(pathComponent)]);`

Reply:
Ahahaa, here we go: `pathComponent` is a number, the index `i` from https://github.com/iov-one/iov-core/blob/v0.6.1/docs/KeyBase.md#simple-address-derivation, so `pathComponent.length` cannot compile. length is always 1.

Reply:
yeah this part i started to change and realized i was going a bit too quickly and probably need to switch to more explicit verbose form to allow easier reading / modification... thanks for help webmaster128 lets just try next time we catch each other online at the same time to finish this quick together. i have the c code pretty well studied but the JS side is slowing me down now on compile errors in TS etc.